File: src/main/scala/com/example/rest/InMemoryPersonRepository.java
package com.example.rest;

import java.util.ArrayList;
import java.util.List;

public class InMemoryPersonRepository implements PersonRepository {

	private final List<Person> persons = new ArrayList<Person>();

	public InMemoryPersonRepository() {
		persons.add(new Person("Eddie", "Merckx"));
		persons.add(new Person("Fausto", "Coppi"));
	}

	@Override
	public List<Person> getAll() {
		return persons;
	}

}

File: src/main/scala/com/example/rest/InMemoryPersonRepository.scala
package com.example.rest

import scala.collection.mutable.{ ListBuffer => List }

class InMemoryPersonRepository extends PersonRepository {
  private val people: List[Person] = List(new Person("Fausto", "Coppi"),
    new Person("Alberto", "Contador"),
    new Person("Miguel", "Indurain"),
    new Person("Eddy", "Merckx"))

  override def getAll() = people

  override def save(person: Person): Unit = {
    people += person
  }
}